### ISSUE TITLE
Tooltips not closing for custom render controls

### ISSUE THREAD

**Issue by NVN-Logis** [dotnet/wpf#7134]
Issue is seen with .NET6.0, including latest 6.0.401
Issue was not present in .NET framework 4.8
Seen on both Windows 10 and 11.

The attached project demonstrates the issue:
If a custom render control does not draw portions of the area that the control covers, then the tooltip should close when the mouse moves off of the drawn parts, as it did in .NET4.8. 
What is observed is that the tooltip stays open as long as the mouse is over the control, including the non-drawn area.

Demonstration: .NET 6.0
![dotnet6 0](https://user-images.githubusercontent.com/114644709/192958631-09d5c75d-a715-4b73-9c1f-d3010914b9bd.gif)

Demonstration: .NET 4.8
![dotnet4 8](https://user-images.githubusercontent.com/114644709/192959314-1d4552f6-7874-4a28-b74c-7445287f2120.gif)

[WpfApp3.zip](https://github.com/dotnet/wpf/files/9671786/WpfApp3.zip)

**Comment by lindexi** [comment]
Reference: https://github.com/dotnet/wpf/pull/6332

It looks like ToolTip is just judging the size of the control but doesn't perform hit testing. 

The size of the TestControl is as big as the window.

![](https://user-images.githubusercontent.com/16054566/194973931-0c1e39d1-0f2f-4a06-a0b7-ec24314765d6.png)

It means that the ToolTip do not close because the ToolTip will think the mouse is always on the control and not disappear.

How about add the Margin to TestControl to test it:

```xml
        <local:TestControl Margin="10,10,10,100" ToolTip="AAAAAAAAAAA" x:Name="Test"/>
```

**Comment by NVN-Logis** [comment]
@lindexi Yes, that is what I think, too. As demonstrated, this is not how it worked in 4.8,

**Comment by skrysmanski** [comment]
@NVN-Logis Could you check - on a machine with the latest Windows patches - if the behavior on .NET 4.8 is still like you'd expect it? I'm seeing a behavior similar to the .NET 6 behavior on my .NET 4.8 application since the last patch day.

**Comment by NVN-Logis** [comment]
@skrysmanski I just updated (Win11) and checked: No change. Still works as expected for .NET 4.8

**Comment by skrysmanski** [comment]
@NVN-Logis Thanks for the feedback. In the meantime I've figure out the cause for my problems (The .NET 4.8.1 runtime breaks WPF tooltips when a progress bar with `IsIndeterminate="True"` is used).

**Comment by Mgamerz** [comment]
This still seems to be an issue; I am having odd behavior with my tooltips not closing, which is causing them to appear to be showing on wrong elements. The behavior is similar to what's shown in the OP. Would be nice to see this fixed.

**Comment by Mgamerz** [comment]
Still an issue on .NET 8, though not sure if it is because the underlying element moved and the hitboxes were not updated (no idea how I would do that).

![2024-06-06_21h23_48](https://github.com/dotnet/wpf/assets/2738836/4d14be93-005f-4fe2-8320-8a20d9b1b145)
